src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,314 @@
+ *
+ * LiquidBounce is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LiquidBounce is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.command.commands.client.fakeplayer
+
+import com.mojang.authlib.GameProfile
+import net.ccbluex.liquidbounce.event.Listenable
+import net.ccbluex.liquidbounce.event.events.GameTickEvent
+import net.ccbluex.liquidbounce.event.events.NotificationEvent
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.features.command.Command
+import net.ccbluex.liquidbounce.features.command.CommandException
+import net.ccbluex.liquidbounce.features.command.builder.CommandBuilder
+import net.ccbluex.liquidbounce.features.command.builder.ParameterBuilder
+import net.ccbluex.liquidbounce.lang.translation
+import net.ccbluex.liquidbounce.utils.client.*
+import net.ccbluex.liquidbounce.utils.combat.getDamageFromExplosion
+import net.minecraft.entity.Entity
+import net.minecraft.network.packet.s2c.play.ExplosionS2CPacket
+import net.minecraft.util.math.Vec3d
+import net.minecraft.world.explosion.Explosion
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.*
+
+/**
+ * Fake Player Command
+ *
+ * Allows you to spawn a client side player for testing purposes.
+ */
+object CommandFakePlayer: Listenable {
+
+    /**
+     * Stores all fake players.
+     */
+    private var fakePlayers: MutableList<FakePlayer>? = null
+
+    private var recording = false
+    private var snapshots: MutableList<PosPoseSnapshot>? = null
+
+    fun createCommand(): Command {
+        return CommandBuilder
+            .begin("fakeplayer")
+            .hub()
+            .subcommand(spawnCommand())
+            .subcommand(removeCommand())
+            .subcommand(clearCommand())
+            .subcommand(startRecordingCommand())
+            .subcommand(endRecordingCommand())
+            .build()
+    }
+
+    private fun spawnCommand() = CommandBuilder
+        .begin("spawn")
+        .parameter(
+            ParameterBuilder
+                .begin<String>("name")
+                .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                .optional()
+                .build()
+        )
+        .handler { _, args ->
+            spawn(args, false)
+        }
+        .build()
+
+    private fun removeCommand() = CommandBuilder
+        .begin("remove")
+        .parameter(
```

Comment:
This function format (`fun foo() = ...`) should only be used for one-line functions. Sure it is possible to write it like this, but it is unreadable.

Reply:
Ah, this was my first kotlin experience and i just used Intellij's extract feature.
This should be a function right?

Reply:
I usually do the same.